### DISCUSSION TITLE
Option to add text or title to overflow menu

### DISCUSSION THREAD

**Discussion by pirupius** [carbon-design-system/carbon#9905] [Help]
Hello everyone, i recently started using the carbon design system about 4 months ago and was in need for a solution i can't find in the documentation. I am in need of an overflow menu as documented here https://carbondesignsystem.com/components/overflow-menu/usage/ with a combination of text as well as an icon. 

It's already possible to render a custom icon with the `renderIcon` attribute which i was able to achieve with `renderIcon={Add16}` but what i need is to add text before the icon to so that the button can change from `+` to `Add +`.

Been looking through some related issues with a suggestion of a custom component being passed to that attribute like so 

```
<OverflowMenu
     flipped={true}
     renderIcon={myCustomIcon}
     >
...
```

but i haven't been able to achieve my goal yet. Has anyone come across a similar need? Any help is appreciated.

**Comment by jnm2377** [comment]
Hey @pirupius unfortunately, we don't currently provide an option to add text to the overflow menu trigger button. If you think this could potentially be a useful design or a common pattern that we're missing, this is definitely a good place to make a case for it. I'm curious if this is a design you have seen in other places and if you would be able to provide those examples (of what you have in mind) for our designers to review.

cc: @carbon-design-system/design
